### PR TITLE
[LMS365] TOPdesk Connector

### PR DESCRIPTION
**Pull request by CJusjong**
---
When submitting a connector, please check the following conditions for your PR to ensure a smooth certification process.

- [X] I attest that the connector works and I verified by deploying and testing all the operations.
- [X] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [X] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [X] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

### CODE REVIEW

File: certified-connectors/TOPdesk/apiDefinition.swagger.json [hunk 1]
```diff
@@ -0,0 +1,11510 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "description": "A Connector for TOPdesk, which entails connectors for Supporting Files, and Incident Management. All endpoint for creating, processing and closing incidents are available in the connector, aswell as alle endpoints for creating and handling branches, locations, persons and operators.",
+    "version": "1.0",
+    "title": "TOPdesk API V1",
```
Comment (sriyen-msft) [inline certified-connectors/TOPdesk/apiDefinition.swagger.json:6]: @CJusjong Please make the title to meet certification [standards](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#give-your-connector-a-title). 

1) Remove capitalization after first letter
2) Remove `API` and `V1` words

File: certified-connectors/TOPdesk/apiDefinition.swagger.json [hunk 2]
```diff
@@ -0,0 +1,11510 @@
...
+      "name": "TOPdesk",
+      "url": "http://www.topdesk.com",
+      "email": "info@topdesk.com"
+    }
+  },
+  "basePath": "/tas/api",
+  "schemes": [
+    "https"
+  ],
+  "x-ms-connector-metadata":[
```
Comment (sriyen-msft) [inline certified-connectors/TOPdesk/apiDefinition.swagger.json:17]: @CJusjong Please fix indentation and alignment issues in all the artifacts

File: certified-connectors/TOPdesk/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,11510 @@
...
+  },
+  {
+  "propertyName": "Privacy Policy",
+  "propertyValue": "https://www.topdesk.com/en/privacy-policy/"
+  },
+  {
+  "propertyName": "Categories",
+  "propertyValue": "Service;Helpdesk;ITSM;IT Operations;ESM"
+  }
+  ]
```
Comment (sriyen-msft) [inline certified-connectors/TOPdesk/apiDefinition.swagger.json:30]: @CJusjong Syntax error - This property should end with a comma. Did you test this connector in your local environment?

File: certified-connectors/TOPdesk/apiDefinition.swagger.json [hunk 4]
```diff
@@ -0,0 +1,11510 @@
...
+        "pageStartAlias": "start",
+        "pageSizeAlias": "page_size",
+        "recommendedPageSize": 5000,
+        "defaultSortField": "id"
+      },
+      "get": {
+        "tags": [
+          "Incident"
+        ],
+        "summary": "Returns a list of incidents.",
```
Comment (sriyen-msft) [inline certified-connectors/TOPdesk/apiDefinition.swagger.json:90]: @CJusjong Summary values should not end with a punctuation mark. Please removeit.

File: certified-connectors/TOPdesk/apiDefinition.swagger.json [hunk 5]
```diff
@@ -0,0 +1,11510 @@
...
+        "description": "| Permission | Remarks |\n| ------- | --------|\n| Operator   |  With read permission on 1st/2nd line incident; Category/Branch/Operator filters apply  |\n| Person   |  Accessible; Person visibility settings apply |\n",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "name": "pageStart",
+            "description": "The offset to start at. The default is 0.\n",
```
Comment (sriyen-msft) [inline certified-connectors/TOPdesk/apiDefinition.swagger.json:98]: @CJusjong Why the property description values are ended with a new line? It is not required, please remove it.

File: certified-connectors/TOPdesk/apiDefinition.swagger.json [hunk 6]
```diff
@@ -0,0 +1,11510 @@
...
+            },
+            "schema": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/Incident"
+              }
+            }
+          },
+          "400": {
+            "description": "_Bad Request_\n<ul>\n  <li>The parameters of the request were malformed</li>\n</ul>\n"
```
Comment (sriyen-msft) [inline certified-connectors/TOPdesk/apiDefinition.swagger.json:169]: @CJusjong The description values of response codes can be a simple text as there are no multiple list items to use html tags.

File: certified-connectors/TOPdesk/apiProperties.json [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+          "tooltip": "Provide the Application Password",
+          "constraints": {
+            "tabIndex": 3,
+            "clearText": false,
+            "required": "true"
+          }
+        }
+      }
+    },
+    "iconBrandColor": "#FFFFFF",
```
Comment (sriyen-msft) [inline certified-connectors/TOPdesk/apiProperties.json:31]: @CJusjong Did you read the checklist mentioned in the PR description ?

image

**Review by sriyen-msft — CHANGES_REQUESTED** [review]

**Comment by sriyen-msft** [comment]
@CJusjong I'm just checking in to see how things are going with updating your connector artifacts as per the given feedback. Please feel free to reach out with any questions you may have, Thanks.

**Comment by sriyen-msft** [comment]
@CJusjong Final Reminder - I'm just checking in to see how things are going with updating your connector artifacts as per the given feedback. Please feel free to reach out with any questions you may have, Thanks.

**Comment by sriyen-msft** [comment]
@CJusjong I am closing this PR as there is no response received from your since long time and you can reopen it whenever you are ready for certification. Thanks

**Comment by CJusjong** [comment]
Hej Srikanth Yenagandhula

Sorry for the missing response. I have been at vacation the last three weeks, why I haven’t responded to the feedback. I will go though the feedback right away, and fix the issues mentioned.

Venlig Hilsen / Best Regards

Christian Kjær Jusjong
Product Specialist
[TOPdesk]
Borupvang 3
2750 Ballerup
+45 2523 7458
+45 7015 7500<tel:+4570157500>
www.topdesk.dk<http://www.topdesk.dk/>

***@***.***<https://blog.topdesk.com/dk>

**Pull request closed without merging**